colcon_bundle/verb/bundle.py

```diff
@@ -159,31 +150,31 @@ def _manage_dependencies(self, context,
 
         installers = self._setup_installers(context)
 
-        print('Collecting dependency information...')
-        jobs = self._get_jobs(context.args, installers, decorators)
-        rc = execute_jobs(context, jobs)
-        if rc != 0:
-            return rc
-
-        print('Fetching and installing dependencies...')
-        installer_metadata = {}
-        for name, installer in installers.items():
-            installer_metadata[name] = installer.install()
-
-        installer_metadata_string = json.dumps(installer_metadata,
-                                               sort_keys=True)
-
-        dependencies_changed = True
-        if os.path.exists(installer_metadata_path):
-            with open(installer_metadata_path, 'r') as f:
-                previous_metadata = f.read()
-                if previous_metadata == installer_metadata_string:
-                    dependencies_changed = False
-
-        with open(installer_metadata_path, 'w') as f:
-            f.write(installer_metadata_string)
-
-        if context.args.include_sources and dependencies_changed:
+        print('Checking if dependencies exist...')
+        logger.info('Checking if dependencies exist...')
+        dependencies_tar_gz_path = os.path.join(
+            bundle_base, 'dependencies.tar.gz')
+
+        if not os.path.exists(dependencies_tar_gz_path):
+            self._check_package_dependency_update(bundle_base, decorators)
+        elif upgrade_deps:
+            self._check_package_dependency_update(bundle_base, decorators)
+            if self._check_installer_dependency_update(
+                    context, decorators, installers, installer_metadata_path):
+                return False
+        else:
+            print('Checking if dependencies have changed since last bundle...')
+            logger.info(
+                'Checking if dependencies have changed since last bundle...')
```

Comment:
Hmm I'm concerned this wording is a bit confusing. The --upgrade arg is "Upgrade dependencies to newest versions if possible", which is using 'dependencies' to mean the entire transitive closure.

Here we are using 'dependencies' to only mean direct dependencies, not the entire dependency closure. I wonder if there's a way to differentiate the two.

Reply:
I changed the parameter to '--upgrade-dependency-graph' and referred concepts as 'local dependencies', 'dependency graph' in the comment/docstring.